File: Team2/Riya_Gupta/ProductOfArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        
        int length = nums.size();
        
        int left[length] , right[length] , j = length -2 ;
        
        left[0] = 1 ; right[length - 1] = 1;
        
        for(int i = 1 ; i < length ; i++)
        {
            left[i] = left[i-1]*nums[i-1];
            
            right[j] = right[j+1]*nums[j+1];
            
            j--;
        }
        
        vector<int> prod ; 
        
        for(int i = 0 ; i  < length ; i++)
        
        prod.push_back(left[i] * right[i]);
        
        return prod ; 
    
    }
};

File: Team1/Riya Tanwar/product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        
        int n = nums.length;
        int[] result = new int[n];
        result[0]=1;
        
        for (int i =0; i<n-1; i++){
            result[i+1]=result[i]*nums[i];
        }
        
        int right =1;
        for (int i=n-1;i>=0;i--){
            result[i]=right*result[i];
            right = right*nums[i];
        }
        return result;

    }
}
